hw/top_earlgrey/ip/ast/ast.core

```diff
@@ -3,13 +3,14 @@ CAPI=2:
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 name: "lowrisc:systems:ast:0.1"
-description: "Analog sensor top (AST) generic views"
-
+description: "Analog Sensor Top generic views"
 filesets:
   files_rtl:
     depend:
       - lowrisc:ip:tlul
+      - lowrisc:ip:pinmux
```

Comment:
ah, I believe this includes an ip-level CSR package which then collides with the corresponding top-level CSR package, and hence causes the CI failure.

can you try switching this to `- lowrisc:ip:pinmux_component`?

Reply:
I tried to to be more exact and put  '- lowrisc:ip:pinmux_pkg' but this cause fusesoc to fail with this error:
WARNING: Unknown item formal in section Target
WARNING: Unknown item formal in section Target
WARNING: Unknown item formal in section Target
ERROR: Conflicting requirements:
Requirements: 'lowrisc_systems_ast == 0.1-0' <- 'lowrisc_ip_pinmux_pkg >= 0-0'
    +lowrisc_systems_ast-0.1-0 was ignored because it depends on missing packages
Requirements: 'lowrisc_systems_ast == 0.1-0'
    Install command rule (+lowrisc_systems_ast-0.1-0)

Failed to resolve dependencies for lowrisc:systems:ast:0.1

I will try '- lowrisc:ip:pinmux_component' now

Reply:
With the '- lowrisc:ip:pinmux_component' if also fails...

Parsing design file '../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux_wkup.sv'

Error-[SV-LCM-PND] Package not defined
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux_wkup.sv, 5
pinmux_wkup, "pinmux_reg_pkg::"
  Package scope resolution failed. Token 'pinmux_reg_pkg' is not a package. 
  Originating module 'pinmux_wkup'.
  Move package definition before the use of the package.


Error-[IND] Identifier not declared
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux_wkup.sv, 19
  Identifier 'WkupCntWidth' has not been declared yet. If this error is not 
  expected, please check if you have set `default_nettype to none.
  
Parsing design file '../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv'
Parsing included file '../src/lowrisc_prim_assert_0.1/rtl/prim_assert.sv'.
Back to file '../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv'.

Error-[SV-LCM-PND] Package not defined
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 10
pinmux, "pinmux_reg_pkg::"
  Package scope resolution failed. Token 'pinmux_reg_pkg' is not a package. 
  Originating module 'pinmux'.
  Move package definition before the use of the package.

Error-[IND] Identifier not declared
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 30
  Identifier 'NMioPeriphOut' has not been declared yet. If this error is not 
  expected, please check if you have set `default_nettype to none.
  
Error-[IND] Identifier not declared
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 32
  Identifier 'NMioPeriphIn' has not been declared yet. If this error is not 
  expected, please check if you have set `default_nettype to none.
  
Error-[IND] Identifier not declared
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 34
  Identifier 'NDioPads' has not been declared yet. If this error is not 
  expected, please check if you have set `default_nettype to none.
  
Error-[IND] Identifier not declared
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 39
  Identifier 'NMioPads' has not been declared yet. If this error is not 
  expected, please check if you have set `default_nettype to none.
 
Error-[SV-EEM-SRE] Scope resolution error
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 56
pinmux, "pinmux_reg_pkg::NMioPeriphOut"
  Target for scope resolution operator does not exist. Token 'pinmux_reg_pkg' 
  is not a class/package.  Originating module 'pinmux'.
  Check that class or package exists with referred token as the name.


Error-[SV-EEM-SRE] Scope resolution error
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 57
pinmux, "pinmux_reg_pkg::NMioPeriphOut"
  Target for scope resolution operator does not exist. Token 'pinmux_reg_pkg' 
  is not a class/package.  Originating module 'pinmux'.
  Check that class or package exists with referred token as the name.


Error-[SV-EEM-SRE] Scope resolution error
../src/lowrisc_ip_pinmux_component_0.1/rtl/pinmux.sv, 58
pinmux, "pinmux_reg_pkg::NDioPads"
  Target for scope resolution operator does not exist. Token 'pinmux_reg_pkg' 
  is not a class/package.  Originating module 'pinmux'.
  Check that class or package exists with referred token as the name.


Note-[MAX_ERROR_COUNT] Maximum error count reached
  Current number of errors has reached the default maximum error count (10).
  Please use +error+<count> to increase the limit.

10 errors

Reply:
```diff
diff --git a/hw/top_earlgrey/ip/ast/ast.core b/hw/top_earlgrey/ip/ast/ast.core
index d0e8be9b8..47d50931d 100644
--- a/hw/top_earlgrey/ip/ast/ast.core
+++ b/hw/top_earlgrey/ip/ast/ast.core
@@ -8,7 +8,8 @@ filesets:
   files_rtl:
     depend:
       - lowrisc:ip:tlul
-      - lowrisc:ip:pinmux
+      - lowrisc:top_earlgrey:pinmux_reg
+      - lowrisc:ip:pinmux_component
       - lowrisc:ip:lc_ctrl_pkg
       - lowrisc:ip:edn_pkg
     files:
```

does the trick for me. This isn't quite right, but good enough for the AST use case I'd presume. (for now at least).

Reply:
It works for me too. Will update the PR with the new ast.core file.

Reply:
@msfschaffner is this an area where it would be better to change the pinmux templating to the fusesoc flag re-direct?

Reply:
yes, the pinmux still follows the "old" style of doing this.
I'll tack this onto the list of pinmux TODOs and change over to the redirect scheme.